infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -221,9 +228,9 @@ public synchronized void handleMessageLayoutPrepare(
         } else {
             // Return the layout with the highest rank proposed before.
             Rank highestProposedRank = proposedLayout == null ? new Rank(-1L, msg.getClientID())
-                    : getPhase2Rank();
-            setPhase1Rank(prepareRank);
-            log.debug("handleMessageLayoutPrepare: New phase 1 rank={}", getPhase1Rank());
+                    : getPhase2Rank(msgEpoch);
```

Comment:
I think the wrappers around the PaxosDataStore methods are superfluous. I thought one of the reasons we factored out those operations was to just consume the PaxosDataStore directly.

Reply:
We will. I have a bigger patch that addresses that. There are other issues with this code that do not necessarily relate to the bug at hand.